pkg/verify/verify.go

```diff
@@ -151,16 +72,28 @@ type ClientBuilder interface {
 	HTTPClient() (*http.Client, error)
```

Comment:
Now that this is on the way to becoming cross-repo API, can we add some docstrings around why this is a function?  Is it intended to be run "occasionally"?  Before every network request?  At least every 5m?  What is the client caching model?